### PR TITLE
Dynamic group subscriptions

### PR DESCRIPTION
**Pull request by marcvberg**
#### Description of changes
Add support for dynamic groups in subscriptions with enhanced filtering

#### Description of how you validated changes
E2E, unit, manual testing

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-auth-transformer/src/__tests__/group-auth.test.ts [hunk 1]
```diff
@@ -291,3 +296,7 @@ test('dynamic group auth generates authorized fields list correctly', () => {
     ## [End] Authorization Steps. **"
   `);
 });
+
+describe('Dynamic group subscription auth tests', () => {
+
```
Comment (AaronZyLee) [inline packages/amplify-graphql-auth-transformer/src/__tests__/group-auth.test.ts:301]: The unit test block is missing

File: packages/graphql-transformers-e2e-tests/src/__tests__/SubscriptionsRuntimeFiltering.e2e.test.ts [hunk 2]
```diff
@@ -505,17 +514,36 @@ test('Multiple owners auth is supported for subscriptions', async () => {
   }) as unknown as Observable<any>;
   let subscription: ZenObservable.Subscription;
   const subscriptionPromise = new Promise((resolve, _) => {
-    subscription = failedObserver.subscribe(
-      event => {},
-      err => {
-        expect(err.error.errors[0].message).toEqual(
-          'Connection failed: {"errors":[{"errorType":"Unauthorized","message":"Not Authorized to access onCreateTaskGroup on type TaskGroup"}]}',
-        );
+    subscription = observer.subscribe(
```
Comment (AaronZyLee) [inline packages/graphql-transformers-e2e-tests/src/__tests__/SubscriptionsRuntimeFiltering.e2e.test.ts:517]: nit: We can also add a test for non-list type dynamic auth since we use different filters (`in` or `containsAny`)

File: packages/amplify-graphql-auth-transformer/src/__tests__/group-auth.test.ts (hunk 1 above)
Comment (phani-srikar) [inline packages/amplify-graphql-auth-transformer/src/__tests__/group-auth.test.ts:301]: Did you want to add more tests here or just an empty block?

File: packages/graphql-transformers-e2e-tests/src/__tests__/SubscriptionsRuntimeFiltering.e2e.test.ts (hunk 2 above)
Comment (phani-srikar) [inline packages/graphql-transformers-e2e-tests/src/__tests__/SubscriptionsRuntimeFiltering.e2e.test.ts:517]: Can we add couple more e2es around:
1. If I am not part of the Dynamic group, I shouldn't be allowed to subscribe.
2. If I have a filter input in addition to dynamic group auth, we appropriately filter the subscriptions.

**Review by phani-srikar — CHANGES_REQUESTED** [review]
Can you please rebase with `main` to resolve the publish error in checks? Thanks.

**Comment by alharris-at** [comment]
Closing this PR in favor of https://github.com/aws-amplify/amplify-category-api/pull/944, which is on a branch multiple folks on the team can iterate on.
